builtin/game/item.lua

```diff
@@ -139,6 +139,19 @@ function core.dir_to_wallmounted(dir)
 	end
 end
 
+function core.wallmounted_to_dir(wallmounted)
+	-- table of dirs in wallmounted order
+	return ({[0]={x=0, y=1, z=0},
+					{x=1, y=-1, z=0},
+					{x=1, y=0, z=0},
+					{x=-1, y=0, z=0},
+					{x=0, y=0, z=1},
+					{x=0, y=0, z=-1}})
+
+					--indexed into by the wallmounted in question
+					[wallmounted]
+end
+
```

Comment:
as far as l know both take such a low amount of memory that it's not noticeable.
You could also cache the table:
define that a local var exists before the function
in the function if this var is not set, set it to the table
take the vector out of it